mint/run/core/aws-sdk-go/quick-tests.go

```diff
@@ -725,6 +725,92 @@ func testListMultipartUploads(s3Client *s3.S3) {
 	successLogger(function, args, startTime).Info()
 }
 
+func testSSECopyObject(s3Client *s3.S3) {
+	// initialize logging params
+	startTime := time.Now()
+	function := "testSSECopyObjectSourceEncrypted"
+	bucketName := randString(60, rand.NewSource(time.Now().UnixNano()), "aws-sdk-go-test-")
+	object := randString(60, rand.NewSource(time.Now().UnixNano()), "")
+	args := map[string]interface{}{
+		"bucketName": bucketName,
+		"objectName": object,
+	}
+	_, err := s3Client.CreateBucket(&s3.CreateBucketInput{
+		Bucket: aws.String(bucketName),
+	})
+	if err != nil {
+		failureLog(function, args, startTime, "", "AWS SDK Go CreateBucket Failed", err).Fatal()
+		return
+	}
+	defer cleanup(s3Client, bucketName, object+"-enc", function, args, startTime, true)
+	defer cleanup(s3Client, bucketName, object+"-noenc", function, args, startTime, false)
+
+	// create encrypted object
+	sseCustomerKey := aws.String("32byteslongsecretkeymustbegiven2")
+	_, errPutEnc := s3Client.PutObject(&s3.PutObjectInput{
+		Body:                 aws.ReadSeekCloser(strings.NewReader("fileToUpload")),
+		Bucket:               aws.String(bucketName),
+		Key:                  aws.String(object + "-enc"),
+		SSECustomerAlgorithm: aws.String("AES256"),
+		SSECustomerKey:       sseCustomerKey,
+	})
+	if errPutEnc != nil {
+		failureLog(function, args, startTime, "", fmt.Sprintf("AWS SDK Go PUT expected to succeed but got %v", err), err).Fatal()
+		return
+	}
+
+	// copy the encrypted object
+	_, errCopyEnc := s3Client.CopyObject(&s3.CopyObjectInput{
+		SSECustomerAlgorithm: aws.String("AES256"),
+		SSECustomerKey:       sseCustomerKey,
+		CopySource:           aws.String(bucketName + "/" + object + "-enc"),
+		Bucket:               aws.String(bucketName),
+		Key:                  aws.String(object + "-copy"),
+	})
+	if errCopyEnc == nil {
+		failureLog(function, args, startTime, "", fmt.Sprintf("AWS SDK Go CopyObject expected to succeed but got %v", err), err).Fatal()
```

Comment:
```suggestion
		failureLog(function, args, startTime, "", fmt.Sprintf("AWS SDK Go CopyObject expected to fail, but it succeeds"), errors.New("Succeeded instead of failing")).Fatal()
```